pkg/util/grpcutil/grpc_util.go

```diff
@@ -103,6 +103,20 @@ func IsAuthError(err error) bool {
 	return false
 }
 
+// IsFailedPreconditionError returns true if err's Cause is an error produced by
+// gRPC due to failed precondition. This error is used internally to shortcut
+// known unavailable nodes and is used by kv retry loop. When outside loop its
```

Comment:
both "known unavailable nodes" "is used by kv retry loop" is ambiguous, can you be more specific? For example

This error is returned when trying to connect to nodes that are decommissioned (and thus will never be reachable again, at least under their old NodeID). The recipient of this error will want to handle this error by either contacting a different node (DistSender does this) when possible, or by returning the error to the client (handling it similar to an AuthError).